crates/swc_ecma_minifier/tests/fixture/next/36127/2/3/output.js

```diff
@@ -117,9 +119,11 @@ class t extends Text {
     }
     replace(i, s, h) {
         if (!(h instanceof t)) return super.replace(i, s, h);
-        let l = n(this.text, n(h.text, n(this.text, [
-            ""
-        ], 0, i)), s), r = this.length + h.length - (s - i);
+        let l = n(this.text, n(h.text, function(t, e, i) {
+            return n(t, [
+                ""
+            ], 0, i);
+        }(this.text, 0, i)), s), r = this.length + h.length - (s - i);
```

Comment:
We should be possible to invoke IIFE in class methods

Reply:
Actually, the key is to determine where we can safely inline declarations. They can be placed inside function scope or block scope – those locations are valid.

However, in places like class fields or anywhere that only accepts expressions, inlining declarations isn't possible. Attempting it there would cause the declarations to be hoisted outside to an enclosing scope, which is risky.